R/standardise_titles.R

```diff
@@ -141,3 +118,30 @@ standardise_titles <- standardize_titles <- function(s, strict = FALSE, api_key
   out <- stringr::str_squish(out)
   out
 }
+
+#' Standardises some words spelling
+#'
+#' Change some words spelling, specifically those that can vary
+#' from one text to another.
+#' @param s A list of character vector
+#' @return A list of character vector with the words changed
+#' @importFrom knitr kable
+#' @examples 
+#' IEADB <- dplyr::slice_sample(qEnviron::agreements$IEADB, n = 10)
+#' IEADB$Title <- standardise_words(IEADB$Title)
+#' @export
+standardise_words <- function(s){
+  # If no arguments, the list of corrected words appears
+  if (missing(s)) {
+    corrected_words <- as.data.frame(corrected_words)
+    corrected_words <- knitr::kable(corrected_words, "simple")
+    corrected_words
+  }  else {
+    # Substitute matching words for corrected words
+    corrected_words <- as.data.frame(corrected_words)
+    for (k in seq_len(nrow(corrected_words))) {
+      s <- gsub(paste0(corrected_words$words[[k]]), paste0(corrected_words$corr_words[[k]]), s, ignore.case = TRUE, perl = T)
+      }
+    s
+  }
+}
```

Comment:
Not sure what's supposed to happen here... Explain/motivate it for me?

Reply:
It used to be a part of `standardise_titles()`, where some words like "co-operation", "south-east" or "wild life" (and 15-20 more words) were changed into "cooperation", "south east" and "wildlife". This operation has been added to the `standardise_titles()` function in order to avoid that treaty titles in different datasets (e.g. IEADB and ECOLEX) were generating false negatives qIDs. Because of different writings, the acronyms generated in the qID were different when it should be the same (co-operation was treated as two words when cooperation as only one). So we added this part to `standardise_titles()`, and in the last PR, you required that this part should be made as a helper function instead of 20 lines of codes.